### PR TITLE
Move EventsAndDenseTriggers lib to ParallelAlgorithms

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

EaDT mirrors EaT very closely so it makes sense that these two should be side-by-side in the directory tree. Unfortunately we can't combine EaDT and EaT because 1) The elliptic code depends on EaT 2) EaDT depends on `Time` 3)  We don't want elliptic code to depend on `Time`. Therefore they have to stay separate. I believe this was the original reason it was put in evolution. Also, I will be using EaDT in my interpolation framework changes, and it felt wrong to include files from the evolution directory in the ParallelAlgorithms directory.

This also moves EaDT out of the `evolution` namespace as well, similar to EaT.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/IO/Observer/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,7 @@ target_link_libraries(
   Utilities
   INTERFACE
   EventsAndTriggers
+  EventsAndDenseTriggers
```
Comment (wthrowe) [inline src/IO/Observer/CMakeLists.txt:48]: Alphabetize.

File: tests/Unit/ParallelAlgorithms/LinearSolver/Gmres/CMakeLists.txt [hunk 2]
```diff
@@ -16,6 +16,7 @@ target_link_libraries(
   Convergence
   DataStructures
   Logging
+  LinearSolverHelpers
```
Comment (wthrowe) [inline tests/Unit/ParallelAlgorithms/LinearSolver/Gmres/CMakeLists.txt:18]: Alphabetize.

File: src/Time/Tags/TimeAndPrevious.hpp [hunk 3]
```diff
@@ -17,9 +17,9 @@
 namespace Tags {
 struct Time;
 }  // namespace Tags
-namespace evolution::Tags {
+namespace Tags {
```
Comment (wthrowe) [inline src/Time/Tags/TimeAndPrevious.hpp]: [optional] Merge the duplicate `Tags` forward-declaration namespaces.  They occur in other files as well.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by knelli2** [comment]
Squashed in the changes because there were only a couple

**Review by wthrowe — APPROVED** [review]

**Pull request merged**